OrbitGl/ThreadTrack.cpp

```diff
@@ -145,9 +145,11 @@ void ThreadTrack::Draw(GlCanvas* canvas, PickingMode picking_mode) {
   TimerTrack::Draw(canvas, picking_mode);
 
   float event_track_height = time_graph_->GetLayout().GetEventTrackHeight();
-  event_track_->SetPos(pos_[0], pos_[1]);
-  event_track_->SetSize(canvas->GetWorldWidth(), event_track_height);
-  event_track_->Draw(canvas, picking_mode);
+  if (thread_id_ != SamplingProfiler::kAllTracepointsFakeTid) {
```

Comment:
Instead of introducing a dependency on SamplingProfiler, maybe we could set a flag on ThreadTrack to control whether or not we want to display event tracks when they are empty. Something like thread_track_->ShowEmptyEventTracks(false);

Reply:
Thank you so much! I eliminated any dependency on SamplingProfiler in ThreadTrack and modified the EventTrack to not show the event bar if there are no callstacks.